daemon/cmd/state.go

```diff
@@ -363,20 +363,22 @@ func (d *Daemon) allocateIPsLocked(ep *endpoint.Endpoint) error {
 	var err error
```

Comment:
For the history, it is more robust for a defer statement to define the variable as the returned value to make even though other operations are done in the middle, the defer is actually executed properly.

```suggestion
func (d *Daemon) allocateIPsLocked(ep *endpoint.Endpoint) (err error) {
    if option.Config.EnableIPv6 && ep.IPv6.IsValid() {
```

Reply:
Fixed in https://github.com/cilium/cilium/pull/25511/commits/6bdaabaffcbf662c87f177ae3894e83725e761e8